File: articles/linear-algebra-for-systems-modeling/articles/what-is-linear-algebra-for-systems-modeling/r/linear_algebra_audit.R
matrix_system <- matrix(
  c(
    0.80, 0.15,
    0.20, 0.90
  ),
  nrow = 2,
  byrow = TRUE
)

model_name <- "two_component_transition_model"
matrix_rank <- qr(matrix_system)$rank
matrix_determinant <- det(matrix_system)
matrix_eigenvalues <- eigen(matrix_system)$values
dominant_eigenvalue <- max(Mod(matrix_eigenvalues))

audit_record <- data.frame(
  model_name = model_name,
  rows = nrow(matrix_system),
  columns = ncol(matrix_system),
  rank = matrix_rank,
  determinant = matrix_determinant,
  dominant_eigenvalue = dominant_eigenvalue,
  matrix_meaning = "transition-like matrix connecting two system components across a modeling step",
  interpretation_warning = paste(
    "Matrix interpretation depends on what entries represent,",
    "how variables are scaled, and whether a linear transformation",
    "is appropriate for the modeled system."
  )
)

dir.create("outputs/tables", recursive = TRUE, showWarnings = FALSE)
write.csv(audit_record, "outputs/tables/r_linear_algebra_matrix_audit.csv", row.names = FALSE)
print(audit_record)

File: articles/linear-algebra-for-systems-modeling/articles/what-is-linear-algebra-for-systems-modeling/julia/linear_algebra_audit.jl
using LinearAlgebra

matrix_system = [0.80 0.15; 0.20 0.90]
rank_value = rank(matrix_system)
determinant_value = det(matrix_system)
eigenvalues = eigvals(matrix_system)
dominant_eigenvalue = maximum(abs.(eigenvalues))

println("model_name,rows,columns,rank,determinant,dominant_eigenvalue,matrix_meaning,interpretation_warning")
println(join((
    "two_component_transition_model",
    size(matrix_system, 1),
    size(matrix_system, 2),
    rank_value,
    determinant_value,
    dominant_eigenvalue,
    "transition-like matrix connecting two system components across a modeling step",
    "Matrix interpretation depends on entries scale and model assumptions."
), ","))
